main.go

```diff
@@ -56,10 +58,54 @@ type cache struct {
 	Metrics []*dto.MetricFamily
 }
 
+func loopUpdateCache() {
+	filter := map[string]string{}
+	for _, item := range strings.Split(opts.Filter, ",") {
+		if !strings.Contains(item, "=") {
+			continue
+		} else {
+			spl := strings.Split(item, "=")
+			filter[spl[0]] = spl[1]
+		}
+	}
+
+	svc := autoscaling.New(session.New())
+	svc.Client.Retryer = client.DefaultRetryer{
+		NumMaxRetries:    9,
+		MinRetryDelay:    1 * time.Second,
+		MinThrottleDelay: 1 * time.Second,
+	}
+
+	for {
+		t := time.Now()
+		data, err := getData(context.Background(), filter, opts.NameTag, svc)
+		logrus.Debugf("Got data in %v seconds", float64(time.Now().Sub(t))/float64(time.Second))
+
+		mu.Lock()
+		if err != nil {
+			logrus.Errorf("Error fetching data: %v", err)
+			if _, ok := errorCounts[err.Error()]; !ok {
+				errorCounts[err.Error()] = 1
+			} else {
+				errorCounts[err.Error()]++
+			}
+		}
+		globalCache = &cache{
+			Date:    time.Now(),
```

Comment:
Could this just use `t` ?

Reply:
If pulling the data takes a long time (many retries, long backoff, whatever), then the data would be considered to be expired as soon as it it put in the cache.